tool/src/org/antlr/v4/codegen/target/GoTarget.java

```diff
@@ -54,6 +57,26 @@ protected void addBadWords() {
 	}
 
 	@Override
+	protected void genFile(Grammar g, ST outputFileST, String fileName) {
+		super.genFile(g, outputFileST, fileName);
+		gofmt(fileName); // consider commenting out when debugging
+	}
+
+	private void gofmt(String fileName) {
+		// If the generated source file is for any reason invalid, gofmt will not change the file and generate output on
+		// stderr. The output of stderr will be lost, gofmt may block asynchronously if the output buffer is not
+		// drained, but in that case gofmt will be killed as soon the JVM terminates. In normal operations the gofmt
+		// process will not get killed prematurely and not create a race condition with the lifecycle of the JVM.
```

Comment:
Yes, but with Java 1.6 and without apache-commons that's very annoying. I've elaborated this comment so much, because I _believe_ this code is in fact not hairy. 

I did check with extreme substitute commands: 

* `("sh", "-c", "cat /dev/zero | tee /tmp/zeros.$$")`: This endlessly copies binary zeros to both Stderr and some files. The buffer is apparently full after exactly 64KiB. Java's documentation says the subprocesses output is then blocked. I have comfirmed these processes _get killed_. 

* `("goyes")` This endlessly copies `y\n` to stdout. I've quickly hacked this myself  with Go, because I wanted to be sure this does not have fancy pipeline signaling like shells do (and I was not sure about "GNU yes").

* `("sh", "-c", "sleep 7272")`: These processes do not produce output and _survive_ the JVM. 

Since `gofmt -w` is a no-op when the source code is invalid, I think all cases are handled.